### PR TITLE
Download missing images

### PR DESCRIPTION
**Pull request by naxoc**
This downloads missing images and is probably re-usable for other sites in the future.

I pulled most code into a new file because there ended up being a lot of it :)

Run it with `wp newspack-content-migrator la-silla-vacia-download-missing-images --media-dir="../path/to/medi" --post-id=5325` and post id is optional.

### CODE REVIEW

File: src/Command/General/DownloadMissingImages.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+			}
+		}
+
+		return false;
+	}
+
+	public function move_local_file( string $from, string $to ): bool {
+		// Make sure the destination folder exists.
+		wp_mkdir_p( dirname( $to ) );
+
```
Comment (eddiesshop) [inline src/Command/General/DownloadMissingImages.php:307]: Can we please log any files that are moved?

File: src/Command/General/DownloadMissingImages.php [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+				wp_update_post( $data['post'] );
+			}
+
+			if ( ! empty( $urls_not_found ) ) {
+				$this->logger->log( $this->log_file,
+					sprintf( "Could not find images for these images for post with ID %d: %s\n\t", $data['post']->ID, implode( "\n\t", $urls_not_found ) ),
+					Logger::WARNING );
+			}
+
+			MigrationMeta::update( $data['post']->ID, $this->command_meta_key, 'post', $this->command_meta_version ); // TODO
```
Comment (eddiesshop) [inline src/Command/General/DownloadMissingImages.php:170]: Not a big deal, but is this TODO still relevant?

**Review by eddiesshop — CHANGES_REQUESTED** [review]
Super clean and efficient! Just 2 very small things and I think we're good to go!

File: src/Command/General/DownloadMissingImages.php (hunk 1 above)
Comment (naxoc) [reply src/Command/General/DownloadMissingImages.php:332]: Yup! I added that

**Comment by naxoc** [comment]
I added one more arg to be used like this: `--post-id-range=1000-20000"` so we can run post id ranges.

**Comment by eddiesshop** [comment]
Tested this out on a few posts and it's working great! Thank you for all this hard work!

**Review by eddiesshop — APPROVED** [review]

**Pull request merged**